lms/djangoapps/save_for_later/helper.py

```diff
@@ -0,0 +1,85 @@
+"""
+helper functions
+"""
+from datetime import datetime
+from django.conf import settings
+from rest_framework.response import Response
+
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.core.djangoapps.catalog.utils import get_programs
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from lms.djangoapps.save_for_later.models import SavedCourse, SavedProgram
+
+
+def get_event_properties(course=None, marketing_url=None, org_img_url=None, program=None):
+    """
+    set event properties for course and program which are required in braze email template
+    """
+    lms_url = configuration_helpers.get_value('LMS_ROOT_URL', settings.LMS_ROOT_URL)
+    event_properties = {
+        'time': datetime.now().isoformat(),
+        'name': 'user.send.save.for.later.email',
+    }
+    if course:
+        event_properties.update({
+            'properties': {
+                'course_image_url': '{base_url}{image_path}'.format(
+                    base_url=lms_url, image_path=course.course_image_url
+                ),
+                'partner_image_url': org_img_url,
+                'enroll_course_url': '{base_url}/register?course_id={course_id}&enrollment_action=enroll&email_opt_in='
+                                     'false&save_for_later=true'.format(base_url=lms_url, course_id=course.id),
+                'view_course_url': marketing_url + '?save_for_later=true' if marketing_url else '#',
+                'display_name': course.display_name,
+                'short_description': course.short_description,
+                'type': 'course',
+            }
+        })
+    if program:
+        event_properties.update({
+            'properties': {
+                'program_image_url': program.get('card_image_url'),
+                'partner_image_url': program.get('authoring_organizations')[0].get('logo_image_url') if program.get(
+                    'authoring_organizations') else None,
+                'view_program_url': program.get('marketing_url') + '?save_for_later=true' if program.get(
+                    'marketing_url') else '#',
+                'title': program.get('title'),
+                'education_level': program.get('type'),
+                'total_courses': len(program.get('courses')) if program.get('courses') else 0,
+                'type': 'program',
+            }
+        })
+    return event_properties
+
+
+def get_course(course_key, user_id, email):
+    """
+    get course overview object and also save course data
+    """
+    try:
+        course_overview = CourseOverview.get_from_id(course_key)
+        SavedCourse.objects.update_or_create(
+            user_id=user_id,
+            email=email,
+            course_id=course_key,
+        )
+        return course_overview
+    except CourseOverview.DoesNotExist:
+        return Response({'error_code': 'course-not-found'}, status=404)
```

Comment:
I don't these will work like this, you need to catch and return these from the view itself.